handler/src/main/java/io/netty/handler/ssl/OpenSslContext.java

```diff
@@ -501,8 +524,52 @@ public OpenSslEngine remove(long ssl) {
         public void add(OpenSslEngine engine) {
             engines.put(engine.sslPointer(), engine);
         }
+
+        @Override
+        public OpenSslEngine get(long ssl) {
+            return engines.get(ssl);
+        }
     }
 
+    static void setKeyMaterial(long ctx, X509Certificate[] keyCertChain, PrivateKey key, String keyPassword)
+            throws SSLException {
+         /* Load the certificate file and private key. */
+        long keyBio = 0;
```

Comment:
sgtm ... bunch of stuff for followup investigation ... plz create followup issues when rdy and we can distribute the work.